app/main.py

```diff
@@ -1,2 +1,136 @@
+from dataclasses import dataclass
+from typing import Optional, Any, List
+
+
+@dataclass
+class Node:
+    key: Any
+    value: Any
+    hash_value: int
+    next: Optional["Node"] = None # noqa
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 0.75
+    ) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = load_factor
+        self.table: List[Optional[Node]] = [None] * self.capacity
```

Comment:
The `__init__` method is correctly implemented with O(n) time complexity, as expected for initialization.